include/tvm/runtime/common_object.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file tvm/runtime/common_object.h
+ * \brief The objects that are commonly used by different runtime, i.e. Relay VM
+ * and interpreter.
+ */
+#ifndef TVM_RUNTIME_COMMON_OBJECT_H_
+#define TVM_RUNTIME_COMMON_OBJECT_H_
+
+#include <tvm/runtime/memory.h>
+#include <tvm/runtime/object.h>
+#include <utility>
+#include <vector>
+
+namespace tvm {
+namespace runtime {
+
+/*!
+ * \brief An object representing a closure. This object is used by both the
+ * Relay VM and interpreter.
+ */
+class ClosureObj : public Object {
```

Comment:
let us still put is into runtime/vm for now, since it does not hurt for Interpreter to use it and this object can is still not self-contained enough to be seen as a Closure(e.g. it cannot be invoked from c++ side)